src/main/java/mate/academy/spring/dao/RoleDao.java

```diff
@@ -0,0 +1,10 @@
+package mate.academy.spring.dao;
+
+import java.util.Optional;
+import mate.academy.spring.model.Role;
+
+public interface RoleDao {
+    Role add(Role role);
+
+    Optional<Role> getByName(String roleName);
```

Comment:
In task description, this method returns Role itself, so I'm not sure if you should use Optional there